force-app/main/default/classes/STG_Panel.cls

```diff
@@ -110,6 +119,61 @@ public with sharing virtual class STG_Panel {
         } private set;
     }
 
+    /**
+    * @description Boolean to determin if the Upgrade to Enhanced RD page should be displayed or not
+    */
+    public Boolean isDisplayingRD2UpgradePage {
+        get {
+            return isDisplayingRD2UpgradePage();
+        } private set;
+    }
+
+    /**
+    * @description Determine if Enhanced RD page should be displayed
+    */
+    private Boolean isDisplayingRD2UpgradePage() {
+        RD2_EnablementDelegate_CTRL.EnablementState enablementState = RD2_EnablementDelegate_CTRL.getEnablementState();
+        if (enablementState.isMetaConfirmed == false) {
+            return true;
+        }
+
+        npe03__Recurring_Donations_Settings__c rdSettings = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();            
+        Map<String, UTIL_BatchJobService.JobSummary> summaryByType = new RD2_DataMigrationEnablement.BatchJob().getSummaryByType(rdSettings);
+        
+        return isMirgrationCompleteInThePast7Days(summaryByType)
+            || isSimpleEnablmentEnableInThePast7Days(summaryByType);
```

Comment:
The way to determine if the upgrade has completed for more than 7 days is by checking if migration batch has ran in the last 7 days/cannot find the apex job from Id (Store in Enablement State), or nightly batch job has ran 7 times if going through simple upgrade.